From eb03e5fec7a11da602a418912a209dd88b273300 Mon Sep 17 00:00:00 2001
From: Jiyong Park <jiyong@google.com>
Date: Tue, 27 Jun 2017 18:29:11 +0900
Subject: [PATCH 09/13] Add missing headers to libbt-vendor

When building with BOARD_VNDK_VERSION set, global include path is not
provided. As a result, libbt-vendor could not be built since it has been
using some symbols without explicitly including headers.

This CL adds the missing headers and organize the include path order.

Bug: 37342627
Test: BOARD_VNDK_VERSION=current m -j libbt-vendor
Change-Id: Icc16cef0aa6bd8911fb2a36bc4016e1f5b110471
---
 libbt-vendor/Android.mk           |  3 +++
 libbt-vendor/include/hci_uart.h   |  1 +
 libbt-vendor/include/hw_ar3k.h    |  2 ++
 libbt-vendor/src/bt_vendor_qcom.c | 26 +++++++++++++-----------
 libbt-vendor/src/hardware.c       | 21 ++++++++++----------
 libbt-vendor/src/hci_smd.c        | 14 +++++++------
 libbt-vendor/src/hci_uart.c       | 13 +++++++-----
 libbt-vendor/src/hw_ar3k.c        | 33 +++++++++++++++----------------
 libbt-vendor/src/hw_rome.c        | 32 ++++++++++++++++--------------
 9 files changed, 80 insertions(+), 65 deletions(-)

diff --git a/libbt-vendor/Android.mk b/libbt-vendor/Android.mk
index cb66f5a..be69144 100644
--- a/libbt-vendor/Android.mk
+++ b/libbt-vendor/Android.mk
@@ -65,6 +65,9 @@ LOCAL_SHARED_LIBRARIES := \
         libcutils \
         liblog
 
+LOCAL_HEADER_LIBRARIES := \
+        libutils_headers
+
 LOCAL_MODULE := libbt-vendor
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE_CLASS := SHARED_LIBRARIES
diff --git a/libbt-vendor/include/hci_uart.h b/libbt-vendor/include/hci_uart.h
index 21e9689..50740d7 100644
--- a/libbt-vendor/include/hci_uart.h
+++ b/libbt-vendor/include/hci_uart.h
@@ -20,6 +20,7 @@
 #define HCI_UART_H
 
 #include <asm-generic/ioctls.h>
+#include <termios.h>
 
 /* Variables to identify the platform */
 /*BT HS UART TTY DEVICE */
diff --git a/libbt-vendor/include/hw_ar3k.h b/libbt-vendor/include/hw_ar3k.h
index 2129548..1b22f4f 100644
--- a/libbt-vendor/include/hw_ar3k.h
+++ b/libbt-vendor/include/hw_ar3k.h
@@ -18,6 +18,8 @@
 #ifndef HW_AR3K_H
 #define HW_AR3K_H
 
+#include <sys/socket.h>
+
 /******************************************************************************
 **  Constants & Macros
 ******************************************************************************/
diff --git a/libbt-vendor/src/bt_vendor_qcom.c b/libbt-vendor/src/bt_vendor_qcom.c
index a8a1866..35df9b0 100644
--- a/libbt-vendor/src/bt_vendor_qcom.c
+++ b/libbt-vendor/src/bt_vendor_qcom.c
@@ -26,22 +26,24 @@
 #define LOG_TAG "bt_vendor"
 #define BLUETOOTH_MAC_ADDR_BOOT_PROPERTY "ro.vendor.boot.btmacaddr"
 
-#include <utils/Log.h>
-#include <cutils/properties.h>
-#include <fcntl.h>
-#include <termios.h>
-#include <unistd.h>
-#include <pthread.h>
+#include "bt_vendor_lib.h"
+#include "bt_vendor_persist.h"
 #include "bt_vendor_qcom.h"
-#include "hci_uart.h"
 #include "hci_smd.h"
+#include "hci_uart.h"
+#include "hw_rome.h"
+
+#include <fcntl.h>
+#include <linux/un.h>
+#include <pthread.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
+#include <termios.h>
+#include <unistd.h>
+
+#include <cutils/properties.h>
 #include <cutils/sockets.h>
-#include <linux/un.h>
-#include "bt_vendor_persist.h"
-#include "hw_rome.h"
-#include "bt_vendor_lib.h"
+#include <utils/Log.h>
 #define WAIT_TIMEOUT 200000
 #define BT_VND_OP_GET_LINESPEED 30
 
@@ -621,7 +623,7 @@ static int init(const bt_vendor_callbacks_t *cb, unsigned char *bdaddr)
 
     temp->rfkill_id = -1;
     temp->enable_extldo = FALSE;
-    temp->cb = cb;
+    temp->cb = (bt_vendor_callbacks_t*)cb;
     temp->ant_fd = -1;
     temp->soc_type = get_bt_soc_type();
     soc_init(temp->soc_type);
diff --git a/libbt-vendor/src/hardware.c b/libbt-vendor/src/hardware.c
index e37e7dd..3df76c4 100644
--- a/libbt-vendor/src/hardware.c
+++ b/libbt-vendor/src/hardware.c
@@ -26,21 +26,22 @@
 
 #define LOG_TAG "bt_vendor"
 
-#include <utils/Log.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <signal.h>
-#include <time.h>
+#include "bt_hci_bdroid.h"
+#include "bt_vendor_qcom.h"
+#include <ctype.h>
+#include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <dirent.h>
-#include <ctype.h>
-#include <cutils/properties.h>
+#include <signal.h>
 #include <stdlib.h>
-#include "bt_hci_bdroid.h"
-#include "bt_vendor_qcom.h"
 #include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
 #include <unistd.h>
+
+#include <cutils/properties.h>
+#include <utils/Log.h>
 #define MAX_CNT_RETRY 100
 
 int hw_config(int nState)
diff --git a/libbt-vendor/src/hci_smd.c b/libbt-vendor/src/hci_smd.c
index ba575b7..63b2a98 100644
--- a/libbt-vendor/src/hci_smd.c
+++ b/libbt-vendor/src/hci_smd.c
@@ -24,17 +24,19 @@
 
 #define LOG_TAG "bt_vendor"
 
-#include <utils/Log.h>
-#include <termios.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <stdio.h>
 #include "bt_vendor_qcom.h"
 #include "hci_smd.h"
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
 #include <string.h>
-#include <cutils/properties.h>
+#include <termios.h>
 #include <unistd.h>
 
+#include <cutils/properties.h>
+#include <utils/Log.h>
+
 /*****************************************************************************
 **   Macros & Constants
 *****************************************************************************/
diff --git a/libbt-vendor/src/hci_uart.c b/libbt-vendor/src/hci_uart.c
index 9e96a13..fa0fc0f 100644
--- a/libbt-vendor/src/hci_uart.c
+++ b/libbt-vendor/src/hci_uart.c
@@ -26,16 +26,19 @@
 
 #define LOG_TAG "bt_vendor"
 
-#include <utils/Log.h>
-#include <termios.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <stdio.h>
 #include "bt_vendor_qcom.h"
 #include "hci_uart.h"
+
+#include <asm-generic/ioctls.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
 #include <string.h>
+#include <termios.h>
 #include <unistd.h>
 
+#include <utils/Log.h>
+
 /******************************************************************************
 **  Constants & Macros
 ******************************************************************************/
diff --git a/libbt-vendor/src/hw_ar3k.c b/libbt-vendor/src/hw_ar3k.c
index 79a385c..edb2118 100644
--- a/libbt-vendor/src/hw_ar3k.c
+++ b/libbt-vendor/src/hw_ar3k.c
@@ -34,29 +34,28 @@ extern "C" {
 
 #define LOG_TAG "bt_vendor"
 
-#include <sys/socket.h>
-#include <utils/Log.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <signal.h>
-#include <time.h>
+#include "bt_hci_bdroid.h"
+#include "bt_vendor_qcom.h"
+#include "hci_uart.h"
+#include "hw_ar3k.h"
+
+#include <ctype.h>
+#include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <dirent.h>
-#include <ctype.h>
-#include <cutils/properties.h>
+#include <signal.h>
 #include <stdlib.h>
-#include <termios.h>
-#include <string.h>
 #include <string.h>
-#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
 #include <sys/uio.h>
+#include <termios.h>
+#include <time.h>
+#include <unistd.h>
 
-#include "bt_hci_bdroid.h"
-#include "bt_vendor_qcom.h"
-#include "hci_uart.h"
-#include "hw_ar3k.h"
-
+#include <cutils/properties.h>
+#include <utils/Log.h>
 /******************************************************************************
 **  Variables
 ******************************************************************************/
diff --git a/libbt-vendor/src/hw_rome.c b/libbt-vendor/src/hw_rome.c
index 5a2b4f5..74b554a 100644
--- a/libbt-vendor/src/hw_rome.c
+++ b/libbt-vendor/src/hw_rome.c
@@ -34,26 +34,28 @@ extern "C" {
 
 #define LOG_TAG "bt_vendor"
 
-#include <sys/socket.h>
-#include <utils/Log.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <signal.h>
-#include <time.h>
+#include "bt_hci_bdroid.h"
+#include "bt_vendor_qcom.h"
+#include "hci_uart.h"
+#include "hw_rome.h"
+
+#include <ctype.h>
+#include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <dirent.h>
-#include <ctype.h>
-#include <cutils/properties.h>
+#include <signal.h>
+#include <stdbool.h>
 #include <stdlib.h>
-#include <termios.h>
 #include <string.h>
-#include <stdbool.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <termios.h>
+#include <time.h>
 #include <unistd.h>
-#include "bt_hci_bdroid.h"
-#include "bt_vendor_qcom.h"
-#include "hci_uart.h"
-#include "hw_rome.h"
+
+#include <cutils/properties.h>
+#include <utils/Log.h>
 
 #define BT_VERSION_FILEPATH "/data/misc/bluedroid/bt_fw_version.txt"
 
-- 
2.17.1

